### PR TITLE
feat: Upgrade dependencies to latest

### PR DESCRIPTION
**Pull request by acezard**
Flagship app navigation feature
Implement app side of AA cozy-bar webview navigation

- [x] Add or upgrade cozy-ui
- [x] Add or upgrade cozy-intent
- [x] Add or upgrade cozy-device-helper
- [x] Upgrade Node CI and .nvmrc to 16 if not already done
- [x] Fix eventual issues caused by the aforementioned upgrades

### CODE REVIEW

File: src/targets/browser/index.jsx [hunk 1]
```diff
@@ -67,12 +68,17 @@ const renderApp = function(appLocale, client, isPublic) {
...
+                  {!isPublic && (
+                    <SharingProvider
+                      doctype="io.cozy.files"
+                      documentType="Notes"
+                    >
+                      <App isPublic={isPublic} />
+                    </SharingProvider>
+                  )}
+                  {isPublic && <App isPublic={isPublic} />}
+                </WebviewIntentProvider>
```
Comment (trollepierre) [inline src/targets/browser/index.jsx:81]: maybe we could test that 🙏

File: src/components/app.jsx [hunk 2]
```diff
@@ -108,9 +111,11 @@ const App = ({ isPublic }) => {
...
         <Layout monoColumn={true}>
           {!isPublic && isMobile && (
-            <BarCenter>
-              <AppTitle>{appName}</AppTitle>
-            </BarCenter>
+            <BarContextProvider webviewService={webviewService}>
+              <BarCenter>
+                <AppTitle>{appName}</AppTitle>
+              </BarCenter>
+            </BarContextProvider>
```
Comment (trollepierre) [inline src/components/app.jsx]: maybe we could test that change to prevent regression 🙏

Comment (acezard) [reply src/components/app.jsx]: actually good catch that shouldn't even be there

File: src/targets/browser/index.jsx (hunk 1 above)
Comment (acezard) [reply src/targets/browser/index.jsx:81]: I'm not sure I see value in testing this, it's too declarative. The provider itself is tested in cozy-intent

File: src/components/app.jsx (hunk 2 above)
Comment (trollepierre) [reply src/components/app.jsx]: I saw 9 commits to review :( I don't know exactly if i should go into details into the whole code...

Comment (acezard) [reply src/components/app.jsx]: oh yeah I see, my bad. Indeed this was removed instead of rebased

**Review by trollepierre — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -61,8 +61,8 @@
     "@material-ui/styles": "4.11.4",
     "classnames": "2.3.1",
     "cozy-bar": "7.16.0",
-    "cozy-client": "27.6.5",
-    "cozy-device-helper": "1.12.0",
+    "cozy-client": "^27.6.5",
+    "cozy-device-helper": "^1.16.1",
```
Comment (Crash--) [inline package.json]: should be squashed with https://github.com/cozy/cozy-notes/pull/260/commits/86b47dec2fdfbef7c4e09efea2ab8c5f00e248e6

File: jest.config.js [hunk 4]
```diff
@@ -12,10 +12,10 @@ module.exports = {
     '^@atlaskit/editor-core(.*)$':
       '<rootDir>/node_modules/cozy-editor-core/src$1'
   },
-  transformIgnorePatterns: ['node_modules/(?!cozy-(ui|sharing))'],
+  transformIgnorePatterns: ['node_modules/(?!cozy-(ui|sharing|harvest-lib))'],
```
Comment (Crash--) [inline jest.config.js:15]: why this harvest-lib change in a commit upgrading cozy-client?!

File: jest.config.js [hunk 5]
```diff
@@ -12,10 +12,10 @@ module.exports = {
...
   transform: {
     '^.+\\.webapp$': '<rootDir>/test/jestLib/json-transformer.js',
-    '^.+\\.(ts|tsx|js|jsx)?$': '<rootDir>/test/jestLib/babel-transformer.js'
+    '^.+\\.(ts|tsx|js|jsx)?$': 'babel-jest'
```
Comment (Crash--) [inline jest.config.js:18]: can you split this commit : one for CC upgrade, one for the jest config please?

File: src/components/app.jsx [hunk 6]
```diff
@@ -111,11 +108,9 @@ const App = ({ isPublic }) => {
...
         <Layout monoColumn={true}>
           {!isPublic && isMobile && (
-            <BarContextProvider webviewService={webviewService}>
-              <BarCenter>
-                <AppTitle>{appName}</AppTitle>
-              </BarCenter>
-            </BarContextProvider>
+            <BarCenter>
+              <AppTitle>{appName}</AppTitle>
+            </BarCenter>
```
Comment (Crash--) [inline src/components/app.jsx:113]: So there is no need to have this commit but also this one https://github.com/cozy/cozy-notes/pull/260/commits/54cd30c3ea07ba6a56e2ebfa0efe86af8b1f3279 

Can you remove them please?

**Review by Crash-- — CHANGES_REQUESTED** [review]
Can you make clear and useful commits please?

**Comment by acezard** [comment]
> Can you make clear and useful commits please?

@Crash-- please refer to this PR https://github.com/cozy/cozy-notes/pull/263

The current PR (260) will only manage the dependency upgrade to latest (node 16, cozy libs, and fix breaking changes). The commit history will be rebased

**Pull request closed without merging**